manifest/services.py

```diff
@@ -280,8 +312,32 @@ def persist(self):
         """
         self.make_paths_relative()
         try:
-            with open(self.path_manifest, 'w') as fmanifest:
+            with open(self.path_manifest, "w") as fmanifest:
                 fmanifest.write(jsonpickle.encode(self.manifest, make_refs=False))
         except EnvironmentError as e:
             self._logger.error(f"COULD NOT write manifest file '{self.path_manifest}'")
-        self._logger.info(f"WROTE manifest file '{self.path_manifest}', session '{self.manifest.session}'")
+        self._logger.info(
+            f"WROTE manifest file '{self.path_manifest}', session '{self.manifest.session}'"
+        )
+
+    def update_manifest_after_run(self) -> None:
```

Comment:
I like the refactoring, as this work is responsibility of the Manifest subsystem, although I would refactor the method name to something like *evaluate_manifest_document*, so it stays more de-coupled from the PIS running process.